Frends.HTTP.Request/Frends.HTTP.Request/Request.cs

```diff
@@ -0,0 +1,400 @@
+               + $":{options.ThrowExceptionOnErrorResponse}:{options.AutomaticCookieHandling}";
+    }
+
+    private static async Task<HttpResponseMessage> GetHttpRequestResponseAsync(
+            HttpClient httpClient, string method, string url,
+            HttpContent content, IDictionary<string, string> headers,
+            Options options, CancellationToken cancellationToken)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+
+        // Only POST, PUT, PATCH and DELETE can have content, otherwise the HttpClient will fail
+        var isContentAllowed = Definitions.Method.TryParse(method, ignoreCase: true, result: out SendMethod _);
+
+        using (var request = new HttpRequestMessage(new HttpMethod(method), new Uri(url))
+        {
+            Content = isContentAllowed ? content : null,
+        })
+        {
+
+            //Clear default headers
+            content.Headers.Clear();
+            foreach (var header in headers)
+            {
+                var requestHeaderAddedSuccessfully = request.Headers.TryAddWithoutValidation(header.Key, header.Value);
+                if (!requestHeaderAddedSuccessfully && request.Content != null)
+                {
+                    //Could not add to request headers try to add to content headers
+                    // this check is probably not needed anymore as the new HttpClient does not seem fail on malformed headers
+                    var contentHeaderAddedSuccessfully = content.Headers.TryAddWithoutValidation(header.Key, header.Value);
+                    if (!contentHeaderAddedSuccessfully)
+                    {
+                        Trace.TraceWarning($"Could not add header {header.Key}:{header.Value}");
+                    }
+                }
+            }
+
+            HttpResponseMessage response;
+            try
+            {
+                response = await httpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
+            }
+            catch (TaskCanceledException canceledException)
+            {
+                if (cancellationToken.IsCancellationRequested)
+                {
+                    // Cancellation is from outside -> Just throw 
+                    throw;
+                }
+
+                // Cancellation is from inside of the request, mostly likely a timeout
+                throw new Exception("HttpRequest was canceled, most likely due to a timeout.", canceledException);
+            }
+
+
+            // this check is probably not needed anymore as the new HttpClient does not fail on invalid charsets
+            if (options.AllowInvalidResponseContentTypeCharSet && response.Content.Headers?.ContentType != null)
+            {
+                response.Content.Headers.ContentType.CharSet = null;
+            }
+
+            return response;
+        }
+    }
+
+    #region Test methods
```

Comment:
Test methods should be moved to unit tests.

Reply:
Checked 'test methods' are needed in main class as well to work properly. Took misdirecting comment and region off.